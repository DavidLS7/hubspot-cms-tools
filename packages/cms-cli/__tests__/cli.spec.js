/*global banjo */

describe('CLI Top Level', () => {
  it('works with help positional', async () =>
    expect(await banjo(['help'])).toMatchSnapshot());

  it('works with help option', async () =>
    expect(await banjo(['--help'])).toMatchSnapshot());

  it('errors with no input', async () => {
    try {
      await banjo();
    } catch (e) {
      expect(e).toMatchInlineSnapshot(`
        "[ERROR] Please specifiy a command or run \`banjo --help\` for a list of available commands
        "
      `);
    }
  });
});
