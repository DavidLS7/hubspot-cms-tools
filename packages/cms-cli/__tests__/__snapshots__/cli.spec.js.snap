// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Top Level works with help option 1`] = `
"Tools for working with the HubSpot CMS

Commands:
  banjo auth [type]                  Configure authentication for a HubSpot
                                     account. Supported authentication protocols
                                     are oauth2 and personalaccesskey.
  banjo init                         initialize hubspot.config.yml for a HubSpot
                                     portal
  banjo logs <path>                  get logs for a function
  banjo lint <path>                  Lint a file or folder for HubL syntax
  banjo hubdb                        Manage HubDB tables
  banjo watch <src> <dest>           Watch a directory on your computer for
                                     changes and upload the changed files to the
                                     HubSpot CMS
  banjo remove <path>                Delete a file or folder from HubSpot
  banjo upload <src> <dest>          Upload a folder or file from your computer
                                     to the HubSpot CMS
  banjo create <type> <name> [dest]  Create HubSpot CMS assets. Supported assets
                                     are function, module, template,
                                     website-theme, react-app, vue-app, and
                                     webpack-serverless.
  banjo fetch <src> [dest]           Fetch a file, directory or module from
                                     HubSpot and write to a path on your
                                     computer
  banjo filemanager                  Commands for working with the File Manager
  banjo secrets                      Manage HubSpot secrets

Options:
  --version    Show version number                                     [boolean]
  --debug, -d  set log level to debug                 [boolean] [default: false]
  --help       Show help                                               [boolean]
"
`;

exports[`CLI Top Level works with help positional 1`] = `
"Tools for working with the HubSpot CMS

Commands:
  banjo auth [type]                  Configure authentication for a HubSpot
                                     account. Supported authentication protocols
                                     are oauth2 and personalaccesskey.
  banjo init                         initialize hubspot.config.yml for a HubSpot
                                     portal
  banjo logs <path>                  get logs for a function
  banjo lint <path>                  Lint a file or folder for HubL syntax
  banjo hubdb                        Manage HubDB tables
  banjo watch <src> <dest>           Watch a directory on your computer for
                                     changes and upload the changed files to the
                                     HubSpot CMS
  banjo remove <path>                Delete a file or folder from HubSpot
  banjo upload <src> <dest>          Upload a folder or file from your computer
                                     to the HubSpot CMS
  banjo create <type> <name> [dest]  Create HubSpot CMS assets. Supported assets
                                     are function, module, template,
                                     website-theme, react-app, vue-app, and
                                     webpack-serverless.
  banjo fetch <src> [dest]           Fetch a file, directory or module from
                                     HubSpot and write to a path on your
                                     computer
  banjo filemanager                  Commands for working with the File Manager
  banjo secrets                      Manage HubSpot secrets

Options:
  --version    Show version number                                     [boolean]
  --debug, -d  set log level to debug                 [boolean] [default: false]
  --help       Show help                                               [boolean]
"
`;
